src/components/MultipleFilter/MultipleFilter.tsx

```diff
@@ -161,10 +161,13 @@ const MultipleFilter = React.forwardRef<HTMLDivElement, MultipleFilterProps>(
         editedFilter.filterCondition;
 
       if (isEdited) {
-        currentReferredFilters[editIndex] = editedFilter;
-        setCurrentReferredFilters(currentReferredFilters.slice());
+        const newReferredFilters = currentReferredFilters.slice();
+        newReferredFilters[editIndex] = editedFilter;
+
+        setCurrentReferredFilters(newReferredFilters);
```

Comment:
現在の値を参照するなら、setCurrentReferredFilters の prevState を使った方がいいかな。
onChange もその中で実行しちゃおう。
ちなみに slice() を実行している理由はなんなんだろう...？新しいオブジェクトとしたいからなのかな。そうだとしたら以下のコードの方がいいと思う。

```suggestion
        setCurrentReferredFilters((prevState) => {
          prevState[editIndex] = editedFilter
          onChange?.(prevState) // 関数に対してもオプショナルチェインが使えるので if 文は不要
          return [...prevState] // 展開することで新しいオブジェクトとして認識させる
        })
```

Reply:
上記でうまく動作しなければこっちかな

```suggestion
        setCurrentReferredFilters((prevState) => {
          const nextState = [...prevState] // 展開することで新しいオブジェクトとして認識させる
          nextState[editIndex] = editedFilter
          onChange?.(nextState) // 関数に対してもオプショナルチェインが使えるので if 文は不要
          return nextState
        })
```

Reply:
@deatiger 
> 上記でうまく動作しなければこっちかな

こちらの方で上手く動作したので、この実装方針で進めていきます！🙇